File: casPratique1/src/main/java/bank/BankServices.java
package bank;

import exceptions.AccountNotFoundException;
import exceptions.InsufficientAmountException;

import java.util.List;


public class BankServices {
    Bank bank = new Bank();

    public BankAccount getBankAccountById(int id) {

        List<BankAccount> list = bank.getBankAccountList();
        for (BankAccount bankAccount : list) {
            if (bankAccount.getId() == id) {
                return bankAccount;
            }
        }
        throw new AccountNotFoundException("Le compte avec l'identifiant: " + id + " n'existe pas ou n'a pas été trouvé.");

    }

    public BankAccount withdraw(BankAccount account, double amount) {
        if (account.getBalance() >= amount) {
            double accountBalance = account.getBalance();
            account.setBalance(accountBalance -= amount);
            return account;
        } else {
            throw new InsufficientAmountException("Montant insuffisant.");
        }
    }
}

File: CasPratique_1/Solution1/EasyBank/BankServices.cs
using System;
using System.Collections.Generic;

namespace EasyBank
{
    public class BankServices
    {
        private List<BankAccount> _bankAccounts;

        public BankServices(List<BankAccount> bankAccounts)
        {
            _bankAccounts = bankAccounts;
        }

        public List<BankAccount> BankAccounts
        {
            get => _bankAccounts;
            set => _bankAccounts = value;
        }
        

        
        public BankAccount GetBankAccountById(int id)
        {
            foreach (var bankAccount in _bankAccounts)
            {
                if (bankAccount.Id == id)
                {
                    return bankAccount;
                }
            }
            throw new ArgumentException("Aucun compte avec l'identifiant renseigné a été trouvé");
        }
    }
}